data/json/recipes/food/offal_dishes.json

```diff
@@ -55,7 +55,10 @@
     "difficulty": 6,
     "charges": 1,
     "time": "30 m",
-    "book_learn": [ [ "mag_glam", 5 ], [ "cookbook_liverforkids", 5, "Buttery Duck Bites" ] ],
+    "book_learn": {
+      "mag_glam": { "skill_level": 5 },
```

Comment:
I'm not sure if using the book item id as the key is a good idea. Usually we put item id together with other data in the same json object, and use a json array to contain all the json objects.

Reply:
I designed it after how it's used in the code. It was a map even before my changes to make sure all entries are unique. This mostly saves the trouble of manually converting everything when reading the JSON. If it's not what you want then I can change it, though I don't really see the problem with it.